homeassistant/components/synology_dsm/__init__.py

```diff
@@ -63,6 +63,10 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.config_entries.async_update_entry(
             entry, data={**entry.data, CONF_VERIFY_SSL: DEFAULT_VERIFY_SSL}
         )
+    if entry.options.get(CONF_TIMEOUT):
+        options = dict(entry.options)
+        options.pop(CONF_TIMEOUT)
+        hass.config_entries.async_update_entry(entry, data=entry.data, options=options)
```

Comment:
We should avoid removing things in the config entry data or options since it makes it impossible to roll back. Just leave the data as it was.

Reply:
#117060